core/src/main/java/org/springframework/security/event/ObservationSecurityEventListener.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.event;
+
+import io.micrometer.observation.Observation;
+import io.micrometer.observation.ObservationRegistry;
+
+import org.springframework.context.ApplicationListener;
+
+/**
+ * A listener for a single security event. Pipes the event to Micrometer's
+ * {@link Observation} API
+ *
+ * @param <T> the type of {@link SecurityEvent}
+ * @author Josh Cummings
+ * @since 6.0
+ */
+public final class ObservationSecurityEventListener<T extends SecurityEvent> implements ApplicationListener<T> {
+
+	private final ObservationRegistry registry;
+
+	private final Observation.KeyValuesProvider<SecurityEventObservationContext<T>> keyValuesProvider;
+
+	/**
+	 * Construct a {@link ObservationSecurityEventListener}
+	 * @param registry the {@link ObservationRegistry} to use
+	 * @param keyValuesProvider the
+	 * {@link io.micrometer.observation.Observation.KeyValuesProvider} to use
+	 */
+	public ObservationSecurityEventListener(ObservationRegistry registry,
+			Observation.KeyValuesProvider<SecurityEventObservationContext<T>> keyValuesProvider) {
+		this.registry = registry;
+		this.keyValuesProvider = keyValuesProvider;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void onApplicationEvent(T event) {
+		String name = "spring.security." + event.getEventType();
+		Observation.Context context = SecurityEventObservationContext.fromEvent(event);
+		Observation.createNotStarted(name, context, this.registry).keyValuesProvider(this.keyValuesProvider)
+				.observe(() -> {
```

Comment:
What is the goal here?
What this will do is measuring the empty code block.

I'm just guessing but I think here signaling that something happened on an existing Observation would make more sense. E.g.:
somebody starts an Observation outside of Spring Security (e.g.: and HTTP request was received) and here we can signal that something happened during that observation:
```java
Observation observation = observationRegistry.getCurrentObservation();
if (observation != null) {
    observation.event(Observation.Event.of("something.happened"));
}
```

Reply:
I think measuring a code block is also a scenario that makes sense for Spring Security, e.g.: instrumenting `AuthenticationProvider` or any component that might need time to do its job, e.g.: using BCrypt, calling out on the network (http, ldap, jdbc, etc), reading files, etc.

Reply:
Excellent! Glad to hear that feature is available now.

Reply:
This is a place where the PR is in flux right now, but because of existing [plans to deprecate `AuthenticationProvider`](https://github.com/spring-projects/spring-security/issues/11428), instrumentation will likely go into [an `AuthenticationManager` implementation](https://github.com/spring-projects/spring-security/pull/11595/files#diff-4f74389b3bef57442d5aa75201d3647ba476b992d66300eecfa74e575bde3629R30) instead of into `AuthenticationProvider`s.

Reply:
Changed to `AuthenticationFilter`, which all the major authentication mechanisms will change over to in the 6.0 release.